src/backend/optimizer/plan/createplan.c

```diff
@@ -2900,10 +2900,6 @@ create_ctescan_plan(PlannerInfo *root, Path *best_path,
 								  scan_relid,
 								  best_path->parent->subplan);
 
-	/* If subplan is not NULL, all costs copied inside make_subqueryscan() */
-	if (!best_path->parent->subplan)
```

Comment:
Here the `best_path->parent->subplan` pointer cannot be NULL, because it is dereferenced before in the `make_subqueryscan` function.